std/algorithm/comparison.d

```diff
@@ -1261,6 +1271,21 @@ MaxType!T max(T...)(T args)
     assert(e == 2);
 }
 
+///
+@safe pure nothrow @nogc unittest
+{
+    import std.math : isNaN;
+    assert(max(2, double.nan).isNaN);
```

Comment:
You could create a loop here to test double, float and real at the same time (this is just a suggestion though).

```
import std.typetuple;
foreach (FloatT; TypeTuple!(float, double, real))
{
    //Code is the same as before
}
```